server/admin/api.go

```diff
@@ -185,16 +186,27 @@ func (s *Server) apiAccountInfo(w http.ResponseWriter, r *http.Request) {
 	writeJSON(w, acctInfo)
 }
 
-// apiBan is the handler for the '/account/{accountID}/ban?rule=RULE' API request.
-func (s *Server) apiBan(w http.ResponseWriter, r *http.Request) {
-	acctIDStr := chi.URLParam(r, accountIDKey)
+// decodeAccountID checks a string as being both hex and the right length and
+// returns its bytes encoded as an account.AccountID.
+func decodeAcctID(acctIDStr string) (account.AccountID, error) {
+	var acctID account.AccountID
 	acctIDSlice, err := hex.DecodeString(acctIDStr)
 	if err != nil {
-		http.Error(w, fmt.Sprintf("could not decode accout id: %v", err), http.StatusBadRequest)
-		return
+		return acctID, fmt.Errorf("could not decode accout id: %v", err)
 	}
 	if len(acctIDSlice) != account.HashSize {
-		http.Error(w, "account id has incorrect length", http.StatusBadRequest)
+		return acctID, errors.New("account id has incorrect length")
+	}
+	copy(acctID[:], acctIDSlice)
```

Comment:
You could also use `n, err := hex.Decode(acctID[:],[]byte(acctIDSlice))` directly too avoid the `copy`, and checking `n` against `account.HashSize`, but if you checked the length first that would be unneeded.